sr_multi_moveit/sr_multi_moveit_config/scripts/generate_robot_srdf.py

```diff
@@ -394,50 +407,94 @@ def parse_arm_groups(self, manipulator_id, manipulator):
-            joint = xml.dom.minidom.Document().createElement(
-                'joint name="' + manipulator.arm.prefix + 'wrist_2_joint" value="1.5708"')
-            new_group.appendChild(joint)
-            joint = xml.dom.minidom.Document().createElement(
-                'joint name="' + manipulator.arm.prefix + 'wrist_3_joint" value="-3.1416"')
-            new_group.appendChild(joint)
-            new_group.writexml(self.new_robot_srdf, indent="  ", addindent="  ", newl="\n")
-        else:
-            joint = xml.dom.minidom.Document().createElement(
-                'joint name="' + manipulator.arm.prefix + 'elbow_joint" value="-2.0"')
-            new_group.appendChild(joint)
-            joint = xml.dom.minidom.Document().createElement(
-                'joint name="' + manipulator.arm.prefix + 'shoulder_lift_joint" value="-1.89"')
-            new_group.appendChild(joint)
-            joint = xml.dom.minidom.Document().createElement(
-                'joint name="' + manipulator.arm.prefix + 'shoulder_pan_joint" value="0.0"')
-            new_group.appendChild(joint)
-            joint = xml.dom.minidom.Document().createElement(
-                'joint name="' + manipulator.arm.prefix + 'wrist_1_joint" value="-2.4"')
-            new_group.appendChild(joint)
-            joint = xml.dom.minidom.Document().createElement(
-                'joint name="' + manipulator.arm.prefix + 'wrist_2_joint" value="-1.5708"')
-            new_group.appendChild(joint)
-            joint = xml.dom.minidom.Document().createElement(
-                'joint name="' + manipulator.arm.prefix + 'wrist_3_joint" value="3.1416"')
-            new_group.appendChild(joint)
-            new_group.writexml(self.new_robot_srdf, indent="  ", addindent="  ", newl="\n")
+    # Get move group states from an XML DOM, returning a dictionary of {move_group: {state: values}
+    @staticmethod
+    def parse_move_group_states(srdf_xml_dom, group_states={}):
+        for group_state_xml in srdf_xml_dom.getElementsByTagName("group_state"):
+            group = group_state_xml.getAttribute("group")
+            state_name = group_state_xml.getAttribute("name")
+            if group not in group_states:
+                group_states[group] = {}
+            group_states[group][state_name] = {}
+            for group_state_child in group_state_xml.childNodes:
+                if group_state_child.localName == "joint":
+                    group_states[group][state_name][group_state_child.getAttribute("name")] = \
+                        group_state_child.getAttribute("value")
+        return group_states
+
+    # Generate multi-robot move group states
+    def add_move_group_states(self):
+        # Check which move groups are in the generated SRDF
+        self.new_robot_srdf.seek(0)
+        move_group_names = list(set(re.findall(r'<group\s+.*name="([^"]*)"', self.new_robot_srdf.read())))
+        # Collect the move group states defined in the seperate robot SRDFs
```

Comment:
typo "separate"

Reply:
Fixed in https://github.com/shadow-robot/sr_interface/pull/768/commits/c17efa1cd76b3ae46e55204f9b9d8b4b765f4c4c.